hazelcast-sql/src/main/java/com/hazelcast/jet/sql/impl/connector/map/OnHeapMapScanP.java

```diff
@@ -0,0 +1,198 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.jet.sql.impl.connector.map;
+
+import com.hazelcast.function.SupplierEx;
+import com.hazelcast.instance.impl.HazelcastInstanceImpl;
+import com.hazelcast.internal.serialization.Data;
+import com.hazelcast.internal.util.collection.PartitionIdSet;
+import com.hazelcast.jet.Traverser;
+import com.hazelcast.jet.core.AbstractProcessor;
+import com.hazelcast.jet.core.Processor;
+import com.hazelcast.jet.core.ProcessorMetaSupplier;
+import com.hazelcast.jet.sql.impl.SimpleExpressionEvalContext;
+import com.hazelcast.map.impl.MapContainer;
+import com.hazelcast.map.impl.MapService;
+import com.hazelcast.spi.impl.NodeEngine;
+import com.hazelcast.sql.impl.QueryUtils;
+import com.hazelcast.sql.impl.exec.scan.KeyValueIterator;
+import com.hazelcast.sql.impl.exec.scan.MapScanExecIterator;
+import com.hazelcast.sql.impl.exec.scan.MapScanRow;
+import com.hazelcast.sql.impl.expression.Expression;
+import com.hazelcast.sql.impl.expression.predicate.TernaryLogic;
+import com.hazelcast.sql.impl.plan.node.MapScanPlanNode;
+import com.hazelcast.sql.impl.row.HeapRow;
+
+import javax.annotation.Nonnull;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.UUID;
+
+/**
+ * This is SQL internal processor to provide a background for DML operations (e.g., SELECT * FROM map).
+ * <p>
+ * The main idea here is to encapsulate all map walking process into IMapTraverser.
+ * There is a special {@link MapScanExecIterator} inside the traverser that allows
+ * it to walk over all map partitions and extract map entries.
+ * IMapTraverser does a setup of MapScanExecIterator and starts traversing map partitions.
+ * Simultaneously, usage of Traverser interface saves simplicity and cleanliness:
+ * it just return all map entries which it can read.
+ */
+public class OnHeapMapScanP extends AbstractProcessor {
+    protected IMapTraverser traverser;
+    private final String mapName;
+    private final MapScanPlanNode planNode;
+
+    public OnHeapMapScanP(
+            @Nonnull String mapName,
+            @Nonnull MapScanPlanNode node
+
+    ) {
+        this.mapName = mapName;
+        this.planNode = node;
+    }
+
+    @Override
+    protected void init(@Nonnull Context context) throws Exception {
+        NodeEngine nodeEngine = ((HazelcastInstanceImpl) context.jetInstance().getHazelcastInstance()).node.nodeEngine;
+        Map<UUID, PartitionIdSet> partitionMap = QueryUtils.createPartitionMap(
```

Comment:
This is not correct. The partition map changes and each processor could see a different snapshot. It can happen that we miss some partition or that two processors will read some partition.